### PR TITLE
feat(misc): add unit interpolation and conversion methods

### PR DESCRIPTION
**Pull request by CodyJasonBennett**
Adds some methods that I commonly use within R3F and my own projects.

### CODE REVIEW

File: packages/maath/src/misc.ts [hunk 1]
```diff
@@ -3,6 +3,53 @@ import { doThreePointsMakeARight } from "./triangle";
...
+  latitude = ((latitude - size.width) / size.width) * -180;
+  longitude = ((longitude - size.height) / size.height) * -90;
+
+  const convertedRadius = Math.cos(degToRad(longitude)) * radius;
+
+  const x = Math.cos(degToRad(latitude)) * convertedRadius;
+  const y = Math.sin(degToRad(longitude)) * radius;
+  const z = Math.sin(degToRad(latitude)) * convertedRadius;
+
+  return { x, y, z };
```
Comment (gsimone) [inline packages/maath/src/misc.ts:50]: We should decide wether to return vectors as {x,y,z} or a tuple [n, n, n]

Comment (CodyJasonBennett) [reply packages/maath/src/misc.ts:50]: Yeah, should I cherry-pick this one into another PR until we sort out math classes?

Comment (gsimone) [reply packages/maath/src/misc.ts:50]: good idea, yep!

File: packages/maath/src/misc.ts [hunk 2]
```diff
@@ -3,6 +3,53 @@ import { doThreePointsMakeARight } from "./triangle";
...
+ * @param latitude Target latitude in degrees.
+ * @param longitude Target longitude in degrees.
+ * @param size Target sphere's dimensions in pixels.
+ * @param radius Target sphere's radius.
+ */
+export function getSphericalCoordinates(
+  latitude: number,
+  longitude: number,
+  size: { width: number; height: number },
+  radius: number
```
Comment (gsimone) [inline packages/maath/src/misc.ts:39]: I'd like to make a common interface for common shapes, eg.

```ts
interface Sphere {
  radius: number,
  center: Vector3
}
```

and use it in sphere-related methods such as the random one. What do you think?

Comment (CodyJasonBennett) [reply packages/maath/src/misc.ts:39]: Good idea. I think I'll wait on adding more 3D stuff until there's a common interface in place.

**Comment by gsimone** [comment]
Yay first contrib, thanks Cody!

**Pull request merged**